bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/communication/GenericCommunicator.java

```diff
@@ -0,0 +1,332 @@
+            return;
+        }
+
+        logger.debug("Step2");
+        // 2: Unknown request (IP150 only)
+        ParadoxIPPacket step2 = new ParadoxIPPacket(ParadoxIPPacket.EMPTY_PAYLOAD, false)
+                .setCommand(HeaderCommand.LOGIN_COMMAND1);
+        sendPacket(step2);
+        receivePacket();
+
+        logger.debug("Step3");
+        // 3: Unknown request (IP150 only)
+        ParadoxIPPacket step3 = new ParadoxIPPacket(ParadoxIPPacket.EMPTY_PAYLOAD, false)
+                .setCommand(HeaderCommand.LOGIN_COMMAND2);
+        sendPacket(step3);
+        receivePacket();
+
+        logger.debug("Step4");
+        // 4: Init communication over UIP softawre request (IP150 and direct serial)
+        byte[] message4 = new byte[37];
+        message4[0] = 0x72;
+        ParadoxIPPacket step4 = new ParadoxIPPacket(message4, true)
+                .setMessageType(HeaderMessageType.SERIAL_PASSTHRU_REQUEST);
+        sendPacket(step4);
+        byte[] receivedPacket = receivePacket();
+        if (receivedPacket != null && receivedPacket.length >= 53) {
+            panelInfoBytes = Arrays.copyOfRange(receivedPacket, 16, 53);
+        }
+
+        logger.debug("Step5");
+        // 5: Unknown request (IP150 only)
+        ParadoxIPPacket step5 = new ParadoxIPPacket(IpMessagesConstants.UNKNOWN_IP150_REQUEST_MESSAGE01, false)
+                .setCommand(HeaderCommand.SERIAL_CONNECTION_INITIATED);
+        sendPacket(step5);
+        receivePacket();
+
+        logger.debug("Step6");
+        // 6: Initialize serial communication request (IP150 and direct serial)
+        byte[] message6 = new byte[37];
+        message6[0] = 0x5F;
+        message6[1] = 0x20;
+        ParadoxIPPacket step6 = new ParadoxIPPacket(message6, true)
+                .setMessageType(HeaderMessageType.SERIAL_PASSTHRU_REQUEST);
+        sendPacket(step6);
+        byte[] response6 = receivePacket();
+        byte[] initializationMessage = Arrays.copyOfRange(response6, 16, response6.length);
+        ParadoxUtil.printPacket("Init communication sub array: ", initializationMessage);
+
+        logger.debug("Step7");
+        // 7: Initialization request (in response to the initialization from the panel)
+        // (IP150 and direct serial)
+        byte[] message7 = generateInitializationRequest(initializationMessage, pcPasswordBytes);
+        ParadoxIPPacket step7 = new ParadoxIPPacket(message7, true)
+                .setMessageType(HeaderMessageType.SERIAL_PASSTHRU_REQUEST).setUnknown0((byte) 0x14);
+        sendPacket(step7);
+        byte[] finalResponse = receivePacket();
+        if ((finalResponse[16] & 0xF0) == 0x10) {
+            logger.info("Successful logon to the panel.");
+            isOnline = true;
+        } else {
+            logger.warn("Logon to panel failure.");
+            logoutSequence();
+        }
+        Thread.sleep(300);
+        // TODO check why after a short sleep, a 37 bytes packet is received after logon

```

Comment:
There is a TODO, what should be done and can it be completed before merging the PR?

Reply:
I don't care about this 37 bytes packet anymore. It works without knowing what exactly it does. :)